plugins/services/src/js/components/ServiceConfigDisplay.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+
+import ConfigurationMap from '../../../../../src/js/components/ConfigurationMap';
+import ConfigurationMapHeading from '../../../../../src/js/components/ConfigurationMapHeading';
+import ConfigurationMapLabel from '../../../../../src/js/components/ConfigurationMapLabel';
+import ConfigurationMapRow from '../../../../../src/js/components/ConfigurationMapRow';
+import ConfigurationMapSection from '../../../../../src/js/components/ConfigurationMapSection';
+import ConfigurationMapValue from '../../../../../src/js/components/ConfigurationMapValue';
+import ServiceConfigDisplayUtil from '../utils/ServiceConfigDisplayUtil';
+import Util from '../../../../../src/js/utils/Util';
+
+const shouldExcludeItem = (row, appDefinition) => {
+  switch (row.key) {
+    case 'gpus':
+      return appDefinition.gpus === 0;
+    case 'container.volumes':
+      return appDefinition.container == null
+        || appDefinition.container.volumes == null
+        || appDefinition.container.volumes.length === 0;
+    case 'healthChecks':
+      return appDefinition.healthChecks == null
+        || appDefinition.healthChecks.length === 0;
+    case 'secrets':
+      return appDefinition.secrets == null
+        || Object.keys(appDefinition.secrets).length === 0;
+    case 'env':
+      return appDefinition.env == null
+        || Object.keys(appDefinition.env).length === 0;
+    case 'labels':
+      return appDefinition.labels == null
+        || Object.keys(appDefinition.labels).length === 0;
+    default:
+      return false;
+  }
+};
+
+const ServiceConfigDisplay = ({appDefinition}) => {
+  const configurationMapSections = ServiceConfigDisplayUtil.displayValues.map(
+    (section, sectionIndex) => {
+      let configurationMapRows = section.values.reduce(
```

Comment:
I agree, this is much more readable. I chose the `map` to reduce the number of times we iterate over items in the array, but I changed it as you suggested. So, done!